cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModel.java

```diff
@@ -1391,4 +1409,16 @@ private void refreshCapacity() {
       _clusterCapacity[r.id()] = capacity;
     }
   }
+
+  private boolean sanityCheckPartitionInfoConsistency(PartitionInfo partitionInfo) {
+    Partition partition = partition(new TopicPartition(partitionInfo.topic(), partitionInfo.partition()));
+    if (partition == null) {
+      return false;
+    }
+
+    Set<Integer> replicasFromPartition = partition.replicas().stream().map(Replica::broker).map(Broker::id).collect(Collectors.toSet());
+    Set<Integer> replicasFromPartitionInfo = Arrays.stream(partitionInfo.replicas()).map(Node::id).collect(Collectors.toSet());
```

Comment:
Converting array to Set and List to Set introduces additional complexity. Under normal circumstances, we expect the order of the replica array and replica list to be the same. Can we first compare the sizes of the list and array and the use a simple loop to compare each element to make sure that they are equal?